models/clm/CAM_bias_correction/CAM4_bias_correct.m

```diff
@@ -0,0 +1,511 @@
+
+% Generate ensemble cell array
+ens_range=[1:ens_mem];
+enstr=cell(1,ens_mem);
+
+for i=1:ens_mem
+    enstr{i}=sprintf('%04d', ens_range(i));
+end
+
+% Check for existence of output file from CAM6_site_grid.sh
+% and set some fixed variables from grid cell collocated with site
+site_grid_file = [path_CAM enstr{1} '/CAM4_NR1.cpl_' enstr{1} '.ha2x1dx6h.' yearstr{1} '.nc'];
+
+if exist(site_grid_file,'file')
+  SITE_doma_area=ncread(site_grid_file,'doma_area');
+  SITE_doma_mask=ncread(site_grid_file,'doma_mask');
+else
+   error('ERROR !! missing site_grid_file, make sure CAM4_site_grid.sh was run:  %s.',site_grid_file)
+end
+
+% Site level met forcing using PLUMBER2 protocol
+% time -->30 min increments in LST (MST)
+% Tair-->Kelvin
+% Qair--> specific humidity (kg kg-1)
+% Wind--> (m/s)
+% SWdown --> (W/m2)
+% LWdown ---> (W/m2)
+% Precip --> (mm/s) or (kg m-2 s-1)
+% Psurf  --> (Pa)
+
+% Load Tower Met Forcing Data
+
+TBOT_main=load_towermet('Tair',yeartower,path_towermet);
+SH_main=load_towermet('Qair',yeartower,path_towermet);
+WIND_main=load_towermet('Wind',yeartower,path_towermet);
+FSDS_main=load_towermet('SWdown',yeartower,path_towermet);
+FLDS_main=load_towermet('LWdown',yeartower,path_towermet);
+PRECTmms_main=load_towermet('Precip',yeartower,path_towermet);
+PSRF_main=load_towermet('Psurf',yeartower,path_towermet);
+YEAR_main=load_towermet('year',yeartower,path_towermet);
+
+
+% Main Loop: Loads CAM, applies correction, writes corrected CAM file
+  for ii = 1:length(yearstr);
+
+    % Time Zones:
+    % Tower Meteorology Forcing (PLUMBER):  LST (MST):  UTC-7
+    % CAM4  Forcing: UTC
+
+    % CAM4 is 'noleap' calendar whereas tower forcing includes leap days
+    % Code accounts for this
+
+    % Syncing time steps:
+    % Tower Forcing (LST) must be advanced 7 hours to synchronize with CAM4 UTC
+    % Total shift of Tower Forcing: +7 hours
+
+
+    % Select only indices of YEAR ii
+    indices=find(YEAR_main==str2num(yearstr{ii}));
+
+    % Pushing met forcing 7 hours forward (go back 14 indices)
+    ta_30=TBOT_main(indices(1)-14:indices(end)-14)';                         % TBOT (Kelvin)
+    q_30=SH_main(indices(1)-14:indices(end)-14)';                            % SH (kg kg-1)
+    wind_30=WIND_main(indices(1)-14:indices(end)-14)';                       % Total Wind (m/s) 
+    sw_30=FSDS_main(indices(1)-14:indices(end)-14)';                         % FSDS (W/m2)
+    lw_30=FLDS_main(indices(1)-14:indices(end)-14)';                         % FLDS (W/m2)
+    ppt_30=PRECTmms_main(indices(1)-14:indices(end)-14)';                    % PRECTmms  (mm/s) or (kg m-2 s-1)
+    ps_30=PSRF_main(indices(1)-14:indices(end)-14)';                         % PSRF (Pa)
+
+    % CAM4 is 'noleap' calendar, thus trim down tower met to 365 days to match
+    % Remove Feb 29th julian day: 60
+
+      if (str2num(yearstr{ii})==2000 | str2num(yearstr{ii})==2004 | str2num(yearstr{ii})==2008)  

```

Comment:
Indenting; likely the same as CAM6_bias_correct.m, so I won't flag them all here.